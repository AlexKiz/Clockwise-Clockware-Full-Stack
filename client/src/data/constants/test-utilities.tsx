import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';


export const renderWithRouter = ({component, path, exact = false}) => (
	render(
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>,
	)
);

export const rerenderWrapper = ({component, path, exact = false}) => {
	return (
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>
	);
};

export * from '@testing-library/react';
