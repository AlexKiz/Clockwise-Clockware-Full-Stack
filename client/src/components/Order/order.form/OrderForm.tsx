/* eslint-disable max-len */
/* eslint-disable react/jsx-key */
import React, {useState, useEffect, FC} from 'react';
import axios from 'axios';
import '../order.form/order-form.css';
import {Master, City, Clock} from '../../../data/types/types';
import {today, openingHours} from '../../../data/constants/systemConstants';
import {OrderFormProps} from './componentConstants';
import {URL} from '../../../data/constants/routeConstants';
import PublicHeader from 'src/components/Headers/PublicHeader';

const OrderForm: FC<OrderFormProps> = () => {
	const [userName, setUserName] = useState<string>('');

	const [userEmail, setUserEmail] = useState<string>('');

	const [orderDate, setOrderDate] = useState<string>('');

	const [orderTime, setOrderTime] = useState<string>(`${openingHours[0]}`);

	const [masterId, setMasterId] = useState<string>('');
	const [masters, setMasters] = useState<Master[]>([]);

	const [cityId, setCityId] = useState<number>(0);
	const [cities, setCities] = useState<City[]>([]);

	const [clockId, setClockId] = useState<number>(0);
	const [clocks, setClocks] = useState<Clock[]>([]);


	useEffect(() => {
		const readCitiesData = async () => {
			const {data} = await axios.get<City[]>(`/${URL.CITY_FOR_ORDER}`);

			if (data.length) {
				setCities(data);
				setCityId(data[0].id);
			}
		};

		readCitiesData();
	}, []);


	useEffect(() => {
		const readClocksData = async () => {
			const {data} = await axios.get<Clock[]>(`/${URL.CLOCK}`);

			if (data.length) {
				setClockId(data[0].id);
				setClocks(data);
			}
		};

		readClocksData();
	}, []);


	useEffect(() => {
		const readAvailableMastersData = async () => {
			if (clocks.length) {
				const {installationTime} = clocks.filter((clock) => clock.id === clockId)[0];
				const endDate = new Date(`${orderDate} ${orderTime}`);
				const startDate = new Date(`${orderDate} ${orderTime}`);
				startDate.setUTCHours(startDate.getHours());
				endDate.setUTCHours(endDate.getHours() + installationTime);

				if (cityId && orderDate && orderTime && clockId) {
					const {data} = await axios.get<Master[]>(`/${URL.AVAILABLE_MASTER}`, {
						params: {
							cityId,
							startWorkOn: startDate.toISOString(),
							endWorkOn: endDate.toISOString(),
						},
					});

					if (!data.length) {
						alert('All masters has been booked at that time. Please choose another time or date');
						setOrderTime('');
						setMasterId('');
						setMasters([]);
					} else {
						setMasterId(data[0].id);
						setMasters(data);
					}
				}
			};
		};
		readAvailableMastersData();
	}, [cityId, clockId, orderDate, orderTime]);


	const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		if (clocks.length) {
			const {installationTime} = clocks.filter((clock) => clock.id === clockId)[0];
			const endDate = new Date(`${orderDate} ${orderTime}`);
			const startDate = new Date(`${orderDate} ${orderTime}`);
			startDate.setUTCHours(startDate.getHours());
			endDate.setUTCHours(endDate.getHours() + installationTime);

			axios.post(`/${URL.ORDER}`,
				{
					name: userName,
					email: userEmail,
					clockId,
					cityId,
					masterId,
					startWorkOn: startDate.toISOString(),
					endWorkOn: endDate.toISOString(),
				});

			setUserName('');
			setUserEmail('');
			setOrderTime('');
			setOrderDate('');
			alert('Your order has been created! Please confirm it on your Emailbox. Have a good day!');
		}
	};

	return (
		<div>
			<PublicHeader/>
			<div className='conteiner'>

				<div className='container-form'>

					<form className='form' onSubmit={onSubmit} name='orderForm'>

						<div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Enter your name:</label>
								</div>

								<input
									type='text'
									placeholder='Ivan Ivanov'
									pattern='^([(A-Za-zА-Яа-я]{3,49})$|^([A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50})$'
									title='User name must be at least 3 letter and alphabetical characters only'
									value={userName}
									onChange={(userNameEvent) => setUserName(userNameEvent.target.value)}
									required
								></input>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Enter your email:</label>
								</div>

								<input
									type='email'
									placeholder='example@mail.com'
									pattern='^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$'
									title='Email must be according the example: myemail@mail.com'
									value={userEmail}
									onChange={(userEmailEvent) => setUserEmail(userEmailEvent.target.value)}
									required
								></input>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Choose clocksize:</label>
								</div>

								<select name='clocksize' onChange={(clockIdEvent) => setClockId(Number(clockIdEvent.target.value))}>
									{
										clocks.map((clock) => (
											<option value={clock.id}>
												{`${clock.size}`}
											</option>
										))
									}
								</select>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Choose your city:</label>
								</div>

								<select name='cities' onChange={(cityIdEvent) => setCityId(Number(cityIdEvent.target.value))}>
									{
										cities.map((city) => (
											<option value={city.id}>
												{`${city.name}`}
											</option>
										))
									}
								</select>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Choose the date:</label>
								</div>

								<input
									type='date'
									name='orderDate'
									min= {today}
									value={orderDate}
									onChange={(orderDateEvent) => setOrderDate(orderDateEvent.target.value)}
								></input>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Choose the time:</label>
								</div>

								<select name='orderTime' onChange={(orderTimeEvent) => setOrderTime(orderTimeEvent.target.value)}>
									{
										openingHours.map((elem) => (
											<option value={elem}>
												{`${elem}`}
											</option>
										))
									}
								</select>
							</div>

							<div className='form-section'>
								<div className='form-input__label'>
									<label>Available masters:</label>
								</div>

								<select name='masterName' onChange={(masterIdEvent) => setMasterId(masterIdEvent.target.value)}>
									{
										masters.map((master) => (
											<option value={master.id}>
												{`${master.name} | Rating:${master.rating.toFixed(2)}`}
											</option>
										))
									}
									<option value="" disabled selected hidden>Choose the master</option>
								</select>
							</div>

							<div className='form-button'>
								<button type="submit"> Create order </button>
							</div>

						</div>

					</form>

				</div>

			</div>
		</div>
	);
};


export default OrderForm;

