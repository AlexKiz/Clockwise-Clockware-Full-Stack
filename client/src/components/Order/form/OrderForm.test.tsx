import React from 'react';
import axios from 'axios';
import OrderForm from './OrderForm';
import {fireEvent, act} from '@testing-library/react';
import {URL} from '../../../data/constants/routeConstants';
import {renderWithRouter, rerenderWrapper} from '../../../data/constants/test-utilities';

const cityForOrder = [{
	id: 1,
	name: 'Dnipro',
}, {
	id: 2,
	name: 'Uzgorod',
}, {
	id: 3,
	name: 'Odessa',
}];

const clocks = [{
	id: 1,
	size: 'Small',
	price: 1,
	installationTime: 1,
}, {
	id: 2,
	size: 'Medium',
	price: 2,
	installationTime: 2,
}, {
	id: 3,
	size: 'Large',
	price: 3,
	installationTime: 3,
}];

const availableMasters = [{
	id: '8f12fab3-5dd6-4a0b-adb2-284766f7a2ce',
	name: 'Super Master',
	rating: 0,
}, {
	id: 'dce55acd-b552-458b-85b5-785e821b336c',
	name: 'Valera',
	rating: 0,
}, {
	id: '234eaf37-740a-47c7-80fe-3a5051dc356b',
	name: 'Master Rybachuk',
	rating: 5,
}];

describe('Order Form Component', () => {
	beforeAll(async () => {
		jest.mock('react-router-dom', () => ({
			...jest.requireActual('react-router-dom'),
			useLocation: () => ({
				pathname: '/',
			}),
		}));

		jest.mock('axios');
	});


	it('matches snapshot', () => {
		const {asFragment} = renderWithRouter({
			component: <OrderForm />,
			path: '/',
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it('shows input with empty value', () => {
		const {getByTestId} = renderWithRouter({
			component: <OrderForm />,
			path: '/',
		});
		expect((getByTestId('user-name-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('user-email-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('clock-size-select') as HTMLInputElement).value).toBe('');
		expect((getByTestId('order-city-select') as HTMLInputElement).value).toBe('');
		expect((getByTestId('order-date-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('order-time-select') as HTMLInputElement).value).toBe('');
		expect((getByTestId('order-master-select') as HTMLInputElement).value).toBe('');
	});

	it('triggers event handlers on inputs and selects', async () => {
		const {getByTestId, rerender} = await renderWithRouter({
			component: <OrderForm />,
			path: '/',
			exact: true,
		});

		axios.mockImplementation(async (url) => {
			switch (url) {
			case URL.CITY_FOR_ORDER:
				return Promise.resolve({
					status: 200,
					data: cityForOrder,
				});

			case URL.CLOCK:
				return Promise.resolve({
					status: 200,
					data: clocks,
				});

			case URL.AVAILABLE_MASTER:
				return Promise.resolve({
					status: 200,
					data: availableMasters,
				});
			}
		});
		act(() => {
			fireEvent.change(getByTestId('user-name-input'), {
				target: {value: 'Test Name'},
			});
			fireEvent.change(getByTestId('user-email-input'), {
				target: {value: 'testEmail@gmail.com'},
			});
			fireEvent.change(getByTestId('clock-size-select'), {
				target: {value: 'Small'},
			});
			fireEvent.change(getByTestId('order-city-select'), {
				target: {value: 'Dnipro'},
			});
			fireEvent.change(getByTestId('order-date-input'), {
				target: {value: '2024-03-23T00:00:00.000Z'},
			});
			fireEvent.change(getByTestId('order-time-select'), {
				target: {value: '09:00'},
			});
			fireEvent.change(getByTestId('order-master-select'), {
				target: {value: 'Some Master'},
			});
		});

		await rerender(rerenderWrapper({
			component: <OrderForm/>,
			path: '/',
			exact: true,
		}));

		expect((getByTestId('user-name-input') as HTMLInputElement).value).toBe('Test Name');
		expect((getByTestId('user-email-input') as HTMLInputElement).value).toBe('testEmail@gmail.com');
		// expect((getByTestId('clock-size-select') as HTMLInputElement).value).toBe('Small');
		// expect((getByTestId('order-city-select') as HTMLInputElement).value).toBe('Dnipro');
		expect((getByTestId('order-date-input') as HTMLInputElement).value).toBe('2024-03-23T00:00:00.000Z');
		expect((getByTestId('order-time-select') as HTMLInputElement).value).toBe('09:00');
		expect((getByTestId('order-time-input') as HTMLInputElement).value).toBe('Some Master');
	});
});
