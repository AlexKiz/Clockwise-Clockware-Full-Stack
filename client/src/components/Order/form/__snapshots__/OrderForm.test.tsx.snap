// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Form Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="wrapper_header"
      >
        <div
          class="wrapper_logo"
        >
          <a
            href="/"
          >
            <div
              class="inner_logo"
            >
              <div
                class="inner_logo_img"
              >
                <div
                  class="logo_img1a"
                >
                  <div
                    class="logo_img1b"
                  >
                    <div
                      class="logo_img1c"
                    />
                  </div>
                </div>
              </div>
              <div
                class="logo_text1d"
              />
            </div>
          </a>
        </div>
        <nav>
          <ul
            class="nav__links"
          >
            <li
              class="active"
            >
              <a
                href="/"
              >
                Make order
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/login"
              >
                Log in
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/registration-form"
              >
                Sign up
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1s4qybw-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  English
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="en"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="conteiner"
    >
      <div
        class="container_form"
      >
        <form
          class="form"
        >
          <div
            class="css-adynvz-MuiStack-root"
          >
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter your name:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="name"
                  id="name-label"
                >
                  Full name
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="user-name-input"
                    id="name"
                    name="name"
                    placeholder="Ivan Ivanov"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter your email:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="user-email-input"
                    id="email"
                    name="email"
                    placeholder="example@mail.com"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Choose clock size:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="clockId"
                >
                  Size
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  required=""
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="clockId clockId"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="clockId"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    data-testid="clock-size-select"
                    name="clockId"
                    required=""
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Size *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Choose your city:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="cityId"
                >
                  City
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  required=""
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="cityId cityId"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="cityId"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    data-testid="order-city-select"
                    name="cityId"
                    required=""
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        City *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Choose the date:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="order-date-input"
                    id="orderDate"
                    min="2022-03-27"
                    name="orderDate"
                    required=""
                    type="date"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Choose the time:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="orderTime"
                >
                  Time
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  required=""
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="orderTime orderTime"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="orderTime"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    data-testid="order-time-select"
                    name="orderTime"
                    required=""
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Time *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Available masters:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="masterId"
                >
                  Choose the master
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  required=""
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="masterId masterId"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="masterId"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    data-testid="order-master-select"
                    name="masterId"
                    required=""
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Choose the master *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="css-1kj7186-MuiStack-root"
              >
                <label
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  for="upload-photo"
                >
                  <input
                    accept=".PNG, .JPG, .JPEG"
                    id="upload-photo"
                    multiple=""
                    name="upload-photo"
                    style="display: none;"
                    type="file"
                  />
                  <span
                    class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <span
                      aria-label="add"
                      class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary css-nubenn-MuiButtonBase-root-MuiFab-root"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                        />
                      </svg>
                       Upload Photos
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-invisible css-fu0map-MuiBadge-badge"
                    />
                  </span>
                </label>
                <span
                  aria-disabled="true"
                  aria-label="add"
                  class="MuiButtonBase-root Mui-disabled MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-default css-lntp72-MuiButtonBase-root-MuiFab-root"
                  role="button"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ImageOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-4.86 8.86-3 3.87L9 13.14 6 17h12l-3.86-5.14z"
                    />
                  </svg>
                   Uploaded Photos
                </span>
              </div>
            </div>
            <div
              class="form_section"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form_btn css-ke5b6m-MuiButtonBase-root-MuiButton-root"
                style="font-size: 18px; border-radius: 15px;"
                tabindex="0"
                type="submit"
              >
                Create Order
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
